pkg/kube/apisix_upstream.go

```diff
@@ -0,0 +1,260 @@
+			return nil, &translateError{
+				field:  "healthCheck.active.unhealthy.tcpFailures",
+				reason: "invalid value",
+			}
+		}
+		active.Unhealthy.TCPFailures = config.Unhealthy.TCPFailures
+		active.Unhealthy.Timeouts = config.Unhealthy.Timeout.Seconds()
+
+		if config.Unhealthy.HTTPCodes != nil && len(config.Unhealthy.HTTPCodes) < 1 {
+			return nil, &translateError{
+				field:  "healthCheck.active.unhealthy.httpCodes",
+				reason: "empty",
+			}
+		}
+		active.Unhealthy.HTTPStatuses = config.Unhealthy.HTTPCodes
+
+		if config.Unhealthy.Interval.Duration < apisixv1.ActiveHealthCheckMinInterval {
+			return nil, &translateError{
+				field:  "healthCheck.active.unhealthy.interval",
+				reason: "invalid value",
+			}
+		}
+		active.Unhealthy.Interval = int(config.Unhealthy.Interval.Seconds())
+	}
+
+	return &active, nil
+}
+
+func (t *translator) translateUpstreamPassiveHealthCheck(config *configv1.PassiveHealthCheck) (*apisixv1.UpstreamPassiveHealthCheck, error) {
+	var passive apisixv1.UpstreamPassiveHealthCheck
+	switch config.Type {
+	case apisixv1.HealthCheckHTTP, apisixv1.HealthCheckHTTPS, apisixv1.HealthCheckTCP:
+		passive.Type = config.Type
+	case "":
+		passive.Type = apisixv1.HealthCheckHTTP
+	default:
+		return nil, &translateError{
+			field:  "healthCheck.passive.Type",
+			reason: "invalid value",
+		}
+	}
+	if config.Healthy != nil {
+		// zero means use the default value.
+		if config.Healthy.Successes < 0 || config.Healthy.Successes > apisixv1.HealthCheckMaxConsecutiveNumber {
+			return nil, &translateError{
+				field:  "healthCheck.passive.healthy.successes",
+				reason: "invalid value",
+			}
+		}
+		passive.Healthy.Successes = config.Healthy.Successes
+		if config.Healthy.HTTPCodes != nil && len(config.Healthy.HTTPCodes) < 1 {
+			return nil, &translateError{
+				field:  "healthCheck.passive.healthy.httpCodes",
+				reason: "empty",
+			}
+		}
+		passive.Healthy.HTTPStatuses = config.Healthy.HTTPCodes
+	}
+
+	if config.Unhealthy != nil {
+		if config.Unhealthy.HTTPFailures < 0 || config.Unhealthy.HTTPFailures > apisixv1.HealthCheckMaxConsecutiveNumber {
+			return nil, &translateError{
+				field:  "healthCheck.passive.unhealthy.httpFailures",
+				reason: "invalid value",
+			}
+		}
+		passive.Unhealthy.HTTPFailures = config.Unhealthy.HTTPFailures
+
+		if config.Unhealthy.TCPFailures < 0 || config.Unhealthy.TCPFailures > apisixv1.HealthCheckMaxConsecutiveNumber {
+			return nil, &translateError{
+				field:  "healthCheck.passive.unhealthy.tcpFailures",
+				reason: "invalid value",
+			}
+		}
+		passive.Unhealthy.TCPFailures = config.Unhealthy.TCPFailures
+		passive.Unhealthy.Timeouts = config.Unhealthy.Timeout.Seconds()
+
+		if config.Unhealthy.HTTPCodes != nil && len(config.Unhealthy.HTTPCodes) < 1 {
+			return nil, &translateError{
+				field:  "healthCheck.passive.unhealthy.httpCodes",
+				reason: "empty",
+			}
+		}
+		passive.Unhealthy.HTTPStatuses = config.Unhealthy.HTTPCodes
+	}
+	return &passive, nil
+}
```

Comment:
We can use `jsonschema` to simplify validation and avoid too many if else. 
This suggest is an enhancement, do not block the PR.

Reply:
We can enhance these logics in the future.